### PR TITLE
refactor: update record type syntax

### LINKED ISSUE #4033: Syntax of Records and Record Types
Why is this inconsistent:

```
let _r: { x :: Int32 } = { x = 123 };
```

Can we change it to:

```
let _r: { x = Int32 } = { x = 123 };
```

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes #4033

### CODE REVIEW

**Comment by jaschdoc** [comment]
@mlutze @JonathanStarup this change requires you update your community build repos.

**Comment by JonathanStarup** [comment]
> @mlutze @JonathanStarup this change requires you update your community build repos.

but this PR can still go through right? Or are they expected to be updated asap?

**Comment by jaschdoc** [comment]
I think that's up to Magnus to decide

**Comment by mlutze** [comment]
For some other things I think we were moving toward "allow both for one version to allow for transition".

**Comment by magnus-madsen** [comment]
Its okay to break the community build, but there is an expectation that those projects will relatively quickly fix their code.

**Comment by magnus-madsen** [comment]
Meanwhile, @jaschdoc :

- There are still some test case failures.
- Will you prepare PRs for doc.flix.dev and for flix.dev that address these syntax changes too?

**Comment by jaschdoc** [comment]
Alright tests are now passing and there's a PR for doc.flix.dev and flix.dev: https://github.com/flix/book/pull/50 and https://github.com/flix/flix.dev/pull/78 respectively

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala [hunk 1]
```diff
@@ -1412,7 +1412,7 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
     }
 
     private def RecordFieldType: Rule1[ParsedAst.RecordFieldType] = rule {
-      SP ~ Names.Field ~ optWS ~ "::" ~ optWS ~ Type ~ SP ~> ParsedAst.RecordFieldType
+      SP ~ Names.Field ~ optWS ~ "=" ~ optWS ~ Type ~ SP ~> ParsedAst.RecordFieldType
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1415]: Here

**Comment by magnus-madsen** [comment]
Let's change the parser to allow both the old and new syntax during a transition period (`"=" || "::"`).

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**